tools/flash_shelly.py

```diff
@@ -0,0 +1,501 @@
+logging.trace = functools.partial(logging.log, logging.TRACE)
+logging.basicConfig(format='%(message)s')
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+arch = platform.system()
+# Windows does not support acsii colours
+WHITE = '\033[1m' if not arch.lower().startswith('win') else ""
+RED = '\033[1;91m' if not arch.lower().startswith('win') else ""
+GREEN = '\033[1;92m' if not arch.lower().startswith('win') else ""
+YELLOW = '\033[1;93m' if not arch.lower().startswith('win') else ""
+BLUE = '\033[1;94m' if not arch.lower().startswith('win') else ""
+PURPLE = '\033[1;95m' if not arch.lower().startswith('win') else ""
+NC = '\033[0m' if not arch.lower().startswith('win') else ""
+
+if not importlib.util.find_spec("zeroconf"):
+  logger.info('Installing zeroconf...')
+  pipe = subprocess.check_output(['pip3', 'install', 'zeroconf'])
+  import zeroconf
+else:
+  import zeroconf
+
+if not importlib.util.find_spec("requests"):
+  logger.info('Installing requests...')
+  pipe = subprocess.check_output(['pip3', 'install', 'requests'])
+  import requests
+else:
+  import requests
+
+class MyListener:
+  def __init__(self):
+    self.device_list = []
+    self.p_list = []
+
+  def add_service(self, zeroconf, type, name):
+      self.device_list.append(name.replace('._http._tcp.local.', ''))
+      # info = zeroconf.get_service_info(type, name, 2000)
+      # logger.trace('INFO: %s' % info)
+      # properties = { y.decode('ascii'): info.properties.get(y).decode('ascii') for y in info.properties.keys() }
+      # self.p_list.append(properties)
+      # logger.trace('properties: %s' % properties)
+      # json_object = json.dumps(properties, indent = 2)
+
+def shelly_model(type, mode):
+  if mode != 'stock':
+    options = {'SHSW-1' : 'Shelly1',
+               'switch1' : 'Shelly1',
+               'SHSW-PM' : 'Shelly1PM',
+               'switch1pm' : 'Shelly1PM',
+               'SHSW-21' : 'Shelly2',
+               'switch2' : 'Shelly2',
+               'SHSW-25' : 'Shelly25',
+               'switch25' : 'Shelly25',
+               'SHPLG-S' : 'ShellyPlugS',
+               'shelly-plug-s' : 'ShellyPlugS',
+               'SHDM-1' : 'ShellyDimmer',
+               'dimmer1' : 'ShellyDimmer',
+               'SHRGBW2' : 'ShellyRGBW2',
+               'rgbw2' : 'ShellyRGBW2',
+    }
+  else:
+    options = {'switch1' : 'SHSW-1',
+               'switch1pm' : 'SHSW-PM',
+               'switch2' : 'SHSW-21',
+               'switch25' : 'SHSW-25',
+               'shelly-plug-s' : 'SHPLG-S',
+               'dimmer1' : 'SHDM-1',
+               'rgbw2' : 'SHRGBW2',
+    }
+  return(options[type])
+
+
+def parseVersion(vs):
+  pp = vs.split("-");
+  v = pp[0].split(".");
+  variant = ""
+  varSeq = 0
+  if len(pp) > 1:
+    i = 0
+    for x in pp[1]:
+      c = pp[1][i]
+      if not c.isnumeric():
+        variant += c
+      else:
+        break
+      i += 1
+    varSeq = int(pp[1][i]) or 0
+  major = int(v[0])
```

Comment:
more Pythonic way:

major, minor, patch = [int for e in v]

Reply:
https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/da9181dd0e2545fbb3295e7b47a0cb81921fea92

Reply:
hmmm did not work.
```
vi1: (<class 'int'>, <class 'int'>, <class 'int'>, '', 0)
vi2: (<class 'int'>, <class 'int'>, <class 'int'>, '', 0)
```